lang/src/org/partiql/lang/syntax/SqlParser.kt

```diff
@@ -2246,6 +2267,135 @@ class SqlParser(private val ion: IonSystem) : Parser {
             else -> err("Expected date string followed by the keyword DATE, found ${head?.value?.type}", PARSE_UNEXPECTED_TOKEN)
         }
 
+    /**
+     * Parses the optional precision specified with TIME type.
+     * The precision states the precision of the second's value in the time unit.
+     * If the precision is specified, the function returns the [ParseNode] with the token as precision value.
+     * Otherwise, the function returns the [ParseNode] with the token as null.
+     * It also verifies that the specified precision is an unsigned integer.
+     */
+    private fun List<Token>.parseOptionalPrecision(): ParseNode =
+        // If the optional precision is present
+        if (head?.type == LEFT_PAREN) {
+            var rem = tail
+            // Expected precision token to be unsigned integer
+            if (rem.head == null || rem.head!!.type != LITERAL || !rem.head!!.value!!.isUnsignedInteger) {
+                rem.head.err("Expected integer value for precision", PARSE_INVALID_PRECISION_FOR_TIME)
```

Comment:
"Expected integer value for precision" -> "Expected integer value between &lt;min> to &lt;max> for precision"?

Reply:
Done. Thanks.